searchlib/src/vespa/searchlib/query/streaming/query.cpp

```diff
@@ -105,73 +120,76 @@ QueryConnector::create(ParseItem::ItemType type)
     }
 }
 
-bool TrueNode::evaluate() const
+bool
+TrueNode::evaluate() const
 {
     return true;
 }
 
-bool AndQueryNode::evaluate() const
+bool
+AndQueryNode::evaluate() const
 {
-  bool ok(true);
-  for (const_iterator it=begin(), mt=end(); ok && (it!=mt); it++) {
-    const QueryNode & qn = **it;
-    ok = ok && qn.evaluate();
-  }
-  return ok;
+    for (const auto & qn : getChildren()) {
+        if ( ! qn->evaluate() ) return false;
+    }
+    return true;
 }
 
-bool AndNotQueryNode::evaluate() const
-{
-  bool ok(empty() ? true : front()->evaluate());
-  if (!empty()) {
-    for (const_iterator it=begin()+1, mt=end(); ok && (it!=mt); it++) {
-      const QueryNode & qn = **it;
-      ok = ok && ! qn.evaluate();
+bool
+AndNotQueryNode::evaluate() const {
+    auto it = getChildren().begin();
+    auto mt = getChildren().end();
+    if ((it != getChildren().end()) && (*it)->evaluate()) {
+        for (++it; it != mt; it++) {
+            if ((*it)->evaluate()) return false;
+        }
+        return true;
     }
-  }
-  return ok;
+    return false;
```

Comment:
I think this should be true, as _ok_ was set to true when empty before (line 125).

Reply:
Fixed